simple/cq_data.c

```diff
@@ -481,7 +481,7 @@ int main(int argc, char **argv)
 
 	hints->ep_attr->type = FI_EP_MSG;
 	hints->caps = FI_MSG;
-	hints->mode = FI_LOCAL_MR | FI_PROV_MR_ATTR;
+	hints->mode = FI_LOCAL_MR | FI_PROV_MR_ATTR | FI_RX_CQ_DATA;
```

Comment:
No - mode bits are things that the provider wants the app to do, not things the app wants.  Apps want CQ data.  The app can specify the size of the data that they require as a domain attribute.  The provider can increase that amount if it wishes.  But CQ data is only ever transferred if the app includes it as part of their protocol.